<<<<<<<< HEAD:patches/wownero/0006-use-proper-error-handling-in-get_seed.patch
From ee6aa49179833c930c1ec10ae2e877aeb87eb8c8 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Mon, 24 Jun 2024 10:49:12 +0200
Subject: [PATCH 06/15] use proper error handling in get_seed
========
From 6be87b2c75ea663e205f468c986809755839afa3 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Mon, 24 Jun 2024 10:49:12 +0200
Subject: [PATCH 05/14] use proper error handling in get_seed
>>>>>>>> e3d763b (cleanup patches (and other stuff) (#79)):patches/monero/0005-use-proper-error-handling-in-get_seed.patch

---
 src/wallet/api/wallet.cpp | 17 ++++++++++++-----
 src/wallet/wallet2.cpp    |  5 ++++-
 2 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/src/wallet/api/wallet.cpp b/src/wallet/api/wallet.cpp
<<<<<<<< HEAD:patches/wownero/0006-use-proper-error-handling-in-get_seed.patch
index 899ef044a..e16d8f83f 100644
========
index 0d85cf3..dfbf4fd 100644
>>>>>>>> e3d763b (cleanup patches (and other stuff) (#79)):patches/monero/0005-use-proper-error-handling-in-get_seed.patch
--- a/src/wallet/api/wallet.cpp
+++ b/src/wallet/api/wallet.cpp
@@ -826,12 +826,19 @@ bool WalletImpl::close(bool store)
 
 std::string WalletImpl::seed(const std::string& seed_offset) const
 {
-    if (checkBackgroundSync("cannot get seed"))
+    if (checkBackgroundSync("cannot get seed")) {
+        setStatusError("cannot get seed");
         return std::string();
-    epee::wipeable_string seed;
-    if (m_wallet)
-        m_wallet->get_seed(seed, seed_offset);
-    return std::string(seed.data(), seed.size()); // TODO
+    }
+    try {
+        epee::wipeable_string seed;
+        if (m_wallet)
+            m_wallet->get_seed(seed, seed_offset);
+        return std::string(seed.data(), seed.size()); // TODO
+    } catch (const std::exception &e) {
+        setStatusError(e.what());
+        return std::string();
+    }
 }
 
 std::string WalletImpl::getSeedLanguage() const
diff --git a/src/wallet/wallet2.cpp b/src/wallet/wallet2.cpp
<<<<<<<< HEAD:patches/wownero/0006-use-proper-error-handling-in-get_seed.patch
index 4e66cdeae..48267d647 100644
--- a/src/wallet/wallet2.cpp
+++ b/src/wallet/wallet2.cpp
@@ -1449,11 +1449,13 @@ bool wallet2::get_seed(epee::wipeable_string& electrum_words, const epee::wipeab
========
index 7913d55..debc0c2 100644
--- a/src/wallet/wallet2.cpp
+++ b/src/wallet/wallet2.cpp
@@ -1440,11 +1440,13 @@ bool wallet2::get_seed(epee::wipeable_string& electrum_words, const epee::wipeab
>>>>>>>> e3d763b (cleanup patches (and other stuff) (#79)):patches/monero/0005-use-proper-error-handling-in-get_seed.patch
   bool keys_deterministic = is_deterministic();
   if (!keys_deterministic)
   {
+    THROW_WALLET_EXCEPTION(error::wallet_internal_error, "This is not a deterministic wallet");
     std::cout << "This is not a deterministic wallet" << std::endl;
     return false;
   }
   if (seed_language.empty())
   {
+    THROW_WALLET_EXCEPTION(error::wallet_internal_error, "seed_language not set");
     std::cout << "seed_language not set" << std::endl;
     return false;
   }
<<<<<<<< HEAD:patches/wownero/0006-use-proper-error-handling-in-get_seed.patch
@@ -1463,8 +1465,9 @@ bool wallet2::get_seed(epee::wipeable_string& electrum_words, const epee::wipeab
========
@@ -1454,8 +1456,9 @@ bool wallet2::get_seed(epee::wipeable_string& electrum_words, const epee::wipeab
>>>>>>>> e3d763b (cleanup patches (and other stuff) (#79)):patches/monero/0005-use-proper-error-handling-in-get_seed.patch
     key = cryptonote::encrypt_key(key, passphrase);
   if (!crypto::ElectrumWords::bytes_to_words(key, electrum_words, seed_language))
   {
+    THROW_WALLET_EXCEPTION(error::wallet_internal_error, "Failed to create seed from key for language: "+seed_language+", falling back to English.");
     std::cout << "Failed to create seed from key for language: " << seed_language << std::endl;
-    return false;
+    crypto::ElectrumWords::bytes_to_words(key, electrum_words, "English");
   }
 
   return true;
-- 
<<<<<<<< HEAD:patches/wownero/0006-use-proper-error-handling-in-get_seed.patch
2.48.0
========
2.39.5 (Apple Git-154)
>>>>>>>> e3d763b (cleanup patches (and other stuff) (#79)):patches/monero/0005-use-proper-error-handling-in-get_seed.patch

